utility_manager.py

```diff
@@ -0,0 +1,147 @@
+
+        #if entry text is a directory, open popup to directory
+        if os.path.isdir(entry_text):
+            file_chooser = FileChooserListView(path=entry_text)        
+        else:
+            file_chooser = FileChooserListView()
+
+        #internal function to select the file
+        def select_file(instance, selection, *args):
+            if selection:
+                # Set selected file path in TextInput
+                file_entry_widget = self.plugin_player.root.ids.file_entry.ids.file_path_input
+                file_entry_widget.text = selection[0] 
+                popup.dismiss()
+        
+        #set up the popup
+        popup = Popup(title='Select a file', content=file_chooser, size_hint=(None, None), size=(500, 500))        
+        file_chooser.bind(on_submit=select_file)
+        popup.open()
+    #load in a new plugin
+    
+        #show a popup to see all class types imported
+    def class_types(self, instance):
+
+        #grab needed info
+        node_number = int(instance.id.split()[0])
+        key_number = int(instance.id.split()[1])
+
+        #start creating the new popup
+        types_box = BoxLayout(orientation='vertical')
+
+        #height of the popup to be updated later to fill whitespace
+        height = 0
+
+        #create a back button
+        #return to main config page if it is a property type
+        if(key_number == -1):
+            back_button = Button(text="Back", size_hint=(None,None), size=(40,20), on_press=self.plugin_player.node_widget_manager.view_config)
+            back_button.id = f'{node_number}'
+        #if its form the add input page, return back there
+        else:
+            back_button = Button(text="Back", size_hint=(None,None), size=(40,20), on_press=self.plugin_player.node_widget_manager.add_input)
+            back_button.id = f'{node_number} {key_number}'
+        types_box.add_widget(back_button)
+        height += 20
+        
+        types_box.add_widget(Label(text="Add a new class type by filling in the blanks separated by a space", color=(0,0,0,1),
+                size_hint_y=None, height=20))
+        
+        height += 20
+
+        
+        #create the entry to add an extra input
+        new_type = BoxLayout(orientation='horizontal', size_hint_y=None, height=35, spacing = 0)
+        new_type_entry = TextInput(hint_text="from _____ import _____", multiline=False, size_hint_x=9/10)
+        new_type.add_widget(new_type_entry)
+
+        #set as global to pull from it when its submitted
+        self.custom_declaration_widget = new_type_entry
+
+        #add an import button
+        add_button = Button(text="Import",size_hint_x = 1/10, on_press=self.new_type)
+        #if its a property type assignt the id to route back to view_config
+        if(key_number == -1):
+            add_button.id = f'{node_number} {-1}'
+        #otherwise route to input page
+        else:
+            add_button.id = f'{node_number} {key_number}'
```

Comment:
```suggestion
        #if its a property type assignt the id to route back to view_config,
        #otherwise route to input page
        add_button.id = f'{node_number} {key_number}'
```
The `id` is set the same way regardless of `key_number` being `-1`.